app/src/main/java/com/sun/premierleague/base/BaseAdapter.kt

```diff
@@ -0,0 +1,62 @@
+package com.sun.premierleague.base
+
+import android.content.Context
+import androidx.recyclerview.widget.RecyclerView
+
+abstract class BaseAdapter<T>(private val collection: List<T>?, private val context: Context) :
+    RecyclerView.Adapter<BaseViewHolder<T>>() {
+
+    private var listItem: MutableList<T>? = null
```

Comment:
data sao có thể null được
t suggest đặt tên list như này nhé 
```suggestion
    private val items = mutableListOf<T>()
```